modules/apps/knowledge-base/knowledge-base-service/src/main/java/com/liferay/knowledge/base/internal/exportimport/data/handler/KBArticleStagedModelDataHandler.java

```diff
@@ -463,12 +465,47 @@ private void _importKBArticleAttachments(
 		serviceContext.setCompanyId(portletDataContext.getCompanyId());
 		serviceContext.setScopeGroupId(portletDataContext.getScopeGroupId());
 
+		Map<String, FileEntry> attachments = new HashMap<>();
+
+		for (FileEntry attachment :
+				importedKBArticle.getAttachmentsFileEntries()) {
+
+			attachments.put(attachment.getUuid(), attachment);
+		}
+
 		for (Element dlFileEntryElement : dlFileEntryElements) {
 			String path = dlFileEntryElement.attributeValue("path");
 
 			FileEntry fileEntry =
 				(FileEntry)portletDataContext.getZipEntryAsObject(path);
 
+			if (attachments.get(fileEntry.getUuid()) != null) {
+				attachments.remove(fileEntry.getUuid());
+
+				continue;
+			}
+
+			DLFileEntry importedFileEntry =
+				_dlFileEntryLocalService.getFileEntryByUuidAndGroupId(
```

Comment:
Maybe you could use com.liferay.portal.kernel.portletfilerepository.PortletFileRepository#fetchPortletFileEntry to get the file entry so we do not have to import the DLFileEntryLocalService in this class

Reply:
I had to use this because I need a UUID-based get/fetch and I did not find one in PortletFileRepository.